### PR TITLE
Qsv to use tiled aligner rather than BLAT

### PR DESCRIPTION
**Pull request by holmeso**
# Description

This PR covers the change in `qsv` from using BLAT to using the tiled aligner (TA).
There are a number of places within qsv where a call is made to BLAT. These have been replaced with a call to the TA.

There are also a number of enhancements to the TA itself after extensive testing against a cohort of 50 melanoma donors.
This was to reduce the number of unique calls to both BLAT and TA.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Additional unit tests have been provided to cover any new methods.
Process has been run against 50 melanoma donors repeatedly.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: q3tiledaligner/src/au/edu/qimr/tiledaligner/util/TARecordUtil.java
Comment (delocalizer) [inline q3tiledaligner/src/au/edu/qimr/tiledaligner/util/TARecordUtil.java:226]: A bit pedantic perhaps, but the name of this implies it returns BLATRecord - is there a reason we're returning the raw String array and not the constructed BLATRecord?

File: q3tiledaligner/src/au/edu/qimr/tiledaligner/util/TiledAlignerUtil.java
Comment (delocalizer) [inline q3tiledaligner/src/au/edu/qimr/tiledaligner/util/TiledAlignerUtil.java:2248]: ndc

File: q3tiledaligner/test/au/edu/qimr/tiledaligner/TARecordTest.java
Comment (delocalizer) [inline q3tiledaligner/test/au/edu/qimr/tiledaligner/TARecordTest.java:186]: ndc

File: q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java
Comment (delocalizer) [inline q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java:1695]: ndc

File: q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java
Comment (delocalizer) [inline q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java:1750]: ndc

File: q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java
Comment (delocalizer) [inline q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java:1806]: ndc

File: q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java
Comment (delocalizer) [inline q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java:1867]: ndc

File: q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java
Comment (delocalizer) [inline q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java:1978]: ndc

File: q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java
Comment (delocalizer) [inline q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java:2037]: ndc

File: q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java
Comment (delocalizer) [inline q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java:2095]: ndc

File: q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java
Comment (delocalizer) [inline q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TARecordUtilTest.java:2331]: ndc

File: q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TiledAlignerUtilTest.java
Comment (delocalizer) [inline q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TiledAlignerUtilTest.java:1007]: ndc

File: q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TiledAlignerUtilTest.java
Comment (delocalizer) [inline q3tiledaligner/test/au/edu/qimr/tiledaligner/util/TiledAlignerUtilTest.java:985]: ndc

File: qcommon/src/org/qcmg/common/sw/SmithWatermanGotoh.java
Comment (delocalizer) [inline qcommon/src/org/qcmg/common/sw/SmithWatermanGotoh.java:203]: ndc

File: qcommon/src/org/qcmg/common/sw/SmithWatermanGotoh.java
Comment (delocalizer) [inline qcommon/src/org/qcmg/common/sw/SmithWatermanGotoh.java:387]: ndc

File: qcommon/src/org/qcmg/common/util/NumberUtils.java
Comment (delocalizer) [inline qcommon/src/org/qcmg/common/util/NumberUtils.java:308]: ndc

File: qsv/src/org/qcmg/qsv/QSVCluster.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/QSVCluster.java:608]: ndc

File: qsv/src/org/qcmg/qsv/QSVCluster.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/QSVCluster.java:697]: ndc

File: qsv/src/org/qcmg/qsv/QSVCluster.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/QSVCluster.java:829]: ndc

File: qsv/src/org/qcmg/qsv/QSVCluster.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/QSVCluster.java:1281]: ndc

File: qsv/src/org/qcmg/qsv/QSVCluster.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/QSVCluster.java:1354]: ndc

File: qsv/src/org/qcmg/qsv/QSVPipeline.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/QSVPipeline.java:504]: ndc

File: qsv/src/org/qcmg/qsv/discordantpair/FindDiscordantPairClustersMT.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/discordantpair/FindDiscordantPairClustersMT.java:187]: ndc

File: qsv/src/org/qcmg/qsv/softclip/Breakpoint.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/softclip/Breakpoint.java:398]: ndc

File: qsv/src/org/qcmg/qsv/softclip/Breakpoint.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/softclip/Breakpoint.java:604]: ndc

File: qsv/src/org/qcmg/qsv/softclip/FindClipClustersMT.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/softclip/FindClipClustersMT.java:1027]: ndc

File: qsv/src/org/qcmg/qsv/softclip/FindClipClustersMT.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/softclip/FindClipClustersMT.java:1109]: ndc

File: qsv/src/org/qcmg/qsv/softclip/SoftClipCluster.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/softclip/SoftClipCluster.java:420]: ndc

File: qsv/src/org/qcmg/qsv/softclip/SoftClipCluster.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/softclip/SoftClipCluster.java:507]: ndc

File: qsv/src/org/qcmg/qsv/splitread/SplitReadContig.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/splitread/SplitReadContig.java:368]: ndc

File: qsv/src/org/qcmg/qsv/splitread/SplitReadContig.java
Comment (delocalizer) [inline qsv/src/org/qcmg/qsv/splitread/SplitReadContig.java:1013]: ndc

File: qsv/test/org/qcmg/qsv/softclip/BreakpointTest.java
Comment (delocalizer) [inline qsv/test/org/qcmg/qsv/softclip/BreakpointTest.java:471]: ndc

File: qsv/test/org/qcmg/qsv/softclip/FindSoftClipClustersMTTest.java
Comment (delocalizer) [inline qsv/test/org/qcmg/qsv/softclip/FindSoftClipClustersMTTest.java:158]: ndc

File: qsv/test/org/qcmg/qsv/softclip/SoftClipClusterTest.java
Comment (delocalizer) [inline qsv/test/org/qcmg/qsv/softclip/SoftClipClusterTest.java:235]: ndc

File: qsv/test/org/qcmg/qsv/splitread/SplitReadContigTest.java
Comment (delocalizer) [inline qsv/test/org/qcmg/qsv/splitread/SplitReadContigTest.java:605]: ndc

**Review by delocalizer — APPROVED** [review]
This is terrific. I feel privileged to be present for this one :)

**Comment by delocalizer** [comment]
@ChristinaXu2017 I would have liked ollie's responses to my questions & comments before merging this.

**Comment by ChristinaXu2017** [comment]
sorry, I just want to merge these changes into my branch, so I can update the regression tests. it is better to leave comments here but not approve If you are waiting for the response.

**Pull request merged**
